realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1698,19 +1698,29 @@ public static boolean compactRealm(RealmConfiguration configuration) {
      * @throws IllegalStateException if called from a non-synchronized (Realm Object Server) Realm.
      */
     @Beta
-    public <E extends RealmModel> void subscribeToObjects(Class<E> clazz, String query, PartialSyncCallback<E> callback) {
+    public <E extends RealmModel> void subscribeToObjects(final Class<E> clazz, String query, final PartialSyncCallback<E> callback) {
         checkIfValid();
         if (!configuration.isSyncConfiguration()) {
             throw new IllegalStateException("Partial sync is only available for synchronized Realm (Realm Object Server)");
         }
 
         sharedRealm.capabilities.checkCanDeliverNotification(BaseRealm.LISTENER_NOT_ALLOWED_MESSAGE);
 
-        Table table = getTable(clazz);
-        String className = table.getClassName();
-        String tableName = table.getName();
+        String className = configuration.getSchemaMediator().getSimpleClassName(clazz);
+        SharedRealm.PartialSyncCallback internalCallback = new SharedRealm.PartialSyncCallback(sharedRealm, className) {
+            @Override
+            public void onSuccess(io.realm.internal.Collection osResults) {
```

Comment:
use import instead of the FQN

Reply:
conflicts with `java.util.Collection` and it is totally my fault ... will rename `io.realm.internal.Collection` to `io.realm.internal.OsResults` in a future PR.